### PR TITLE
MINOR: Increase some unit tests for utils

### PR DESCRIPTION
**Pull request by xijiu**
Hello, I noticed that some methods in the `Utils.java` class lack corresponding unit tests when I was reading the code, so added them.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by m1a2st — COMMENTED** [review]
@xijiu, Thanks for your comments, I leaved some comments

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 1]
```diff
@@ -448,6 +449,87 @@ public void testMin() {
...
+    @Test
+    public void mkStringTest() {
+        Map<String, String> map = new LinkedHashMap<>();
+        map.put("key1", "val1");
+        map.put("key2", "val2");
+        map.put("key3", "val3");
+        String result = Utils.mkString(map, "__begin__", "__end__", "=", ",");
+        assertEquals("__begin__key1=val1,key2=val2,key3=val3__end__", result);
+
+        String result2 = Utils.mkString(new HashMap<>(), "__begin__", "__end__", "=", ",");
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: `Collections.emptyMap()` is better

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 2]
```diff
@@ -448,6 +449,87 @@ public void testMin() {
...
+        map.clear();
+        map.put("key4", "val4");
+        map.put("key5", "val5");
+        String result4 = Utils.mkString(map, null, null, " : ", ";");
+        assertEquals("key4 : val4;key5 : val5", result4);
+
+        map.clear();
+        map.put("key6", "val6");
+        String result5 = Utils.mkString(map, null, null, " : ", ";");
+        assertEquals("key6 : val6", result5);
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: Could you add a scenario for `keyValueSeparator` or `elementSeparator` is null

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 3]
```diff
@@ -448,6 +449,87 @@ public void testMin() {
...
+    @Test
+    public void parseMapTest() {
+        Map<String, String> map1 = Utils.parseMap("k1=v1,k2=v2,k3=v3", "=", ",");
+        assertEquals(3, map1.size());
+        assertTrue(map1.containsKey("k1"));
+        assertTrue(map1.containsKey("k2"));
+        assertTrue(map1.containsKey("k3"));
+        assertTrue(map1.containsValue("v1"));
+        assertTrue(map1.containsValue("v2"));
+        assertTrue(map1.containsValue("v3"));
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: I think to test `k1` key get value is equals to `v2` is more better, WDTY

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 4]
```diff
@@ -448,6 +449,87 @@ public void testMin() {
...
+
+        Map<String, String> map2 = Utils.parseMap(null, "=", ",");
+        assertEquals(0, map2.size());
+
+        Map<String, String> map3 = Utils.parseMap("k4=v4,k5=v5=vv5=vvv5", "=", ",");
+        assertEquals(2, map3.size());
+        assertTrue(map3.containsKey("k4"));
+        assertTrue(map3.containsKey("k5"));
+        assertTrue(map3.containsValue("v4"));
+        assertTrue(map3.containsValue("v5=vv5=vvv5"));
```
Comment (m1a2st) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: I think to test `k5` key get value is equals to`v5=vv5=vvv5` is more better, WDTY

**Review by m1a2st — COMMENTED** [review]
@xijiu, Thanks for your comments, I leaved some comments

**Comment by xijiu** [comment]
> @xijiu, Thanks for your comments, I leaved some comments

@m1a2st 
Thanks for your code review. They are all very helpful.  I have fixed all of them, please review again.

**Review by m1a2st — APPROVED** [review]
LGTM, @chia7712, PTAL, Thank you

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**